src/core/rebalance/src/4C_rebalance_graph_based.cpp

```diff
@@ -353,8 +354,16 @@ std::shared_ptr<const Core::LinAlg::Graph> Core::Rebalance::build_monolithic_nod
   std::vector<std::pair<int, Core::GeometricSearch::BoundingVolume>> bounding_boxes;
   for (auto element : dis.my_row_element_range())
   {
-    bounding_boxes.emplace_back(std::make_pair(element.global_id(),
-        element.user_element()->get_bounding_volume(dis, zero_vector, params)));
+    if (displacement == nullptr)
+    {
+      bounding_boxes.emplace_back(std::make_pair(element.global_id(),
+          element.user_element()->get_bounding_volume(dis, zero_vector, params)));
```

Comment:
The `zero_vector` is only needed here, so its creation should be moved inside this `if` branch.

Reply:
I moved the creation of the zero vector into the if branch.